server/app/services/export/CsvExporter.java

```diff
@@ -168,49 +162,14 @@ public void exportRecord(
    * from the raw value in applicant data, such as translating enum number to human-readable text in
    * default locale or mapping file key to download url.
    */
-  private String getValueFromAnswerMap(Column column, ImmutableMap<Path, AnswerData> answerMap) {
-    Path path = column.jsonPath().orElseThrow();
-    if (!answerMap.containsKey(path)) {
+  private String getValueFromQuestionMap(
+      Column column, ImmutableMap<Path, ApplicantQuestion> questionMap) {
+    Path path = column.questionPath().orElseThrow();
+    if (!questionMap.containsKey(path)) {
       return EMPTY_VALUE;
     }
 
-    var answerData = answerMap.get(path);
-    // if it's not a checkbox question, get the answer out of AnswerData
-    if (!answerData.questionDefinition().getQuestionType().equals(QuestionType.CHECKBOX)) {
-      return answerData.scalarAnswersInDefaultLocale().get(path);
-    }
-
-    // if it is a checkbox question, use the optionAdminName in the Column to fill the column.
-    if (column.optionAdminName().isEmpty()) {
-      throw new IllegalStateException(
-          "CSV Column represents a checkbox question, but no option admin name is specified.");
-    }
-
-    if (!answerData.isAnswered()) {
-      return MultiOptionSelectionExportType.NOT_ANSWERED.toString();
-    }
-
-    ImmutableList<String> selectedList =
-        answerData
-            .applicantQuestion()
-            .createMultiSelectQuestion()
-            .getSelectedOptionAdminNames()
-            .orElse(ImmutableList.of());
-
-    ImmutableList<String> allOptionsShownInQuestionVersion =
-        answerData.applicantQuestion().createMultiSelectQuestion().getOptions().stream()
-            .map(LocalizedQuestionOption::adminName)
-            .collect(ImmutableList.toImmutableList());
-
-    if (!allOptionsShownInQuestionVersion.contains(column.optionAdminName().get())) {
-      return MultiOptionSelectionExportType.NOT_AN_OPTION_AT_PROGRAM_VERSION.toString();
-    }
-
-    if (selectedList.contains(column.optionAdminName().get())) {
-      return MultiOptionSelectionExportType.SELECTED.toString();
-    }
-
-    return MultiOptionSelectionExportType.NOT_SELECTED.toString();
+    return column.answerExtractor().get().apply(questionMap.get(path).getQuestion());
```

Comment:
suggestion (non-blocking): I think a comment here could be helpful since this is where the magic happens.

Reply:
done!